src/sentry/tasks/statistical_detectors.py

```diff
@@ -96,10 +99,13 @@ def run_detection() -> None:
         Project.objects.filter(status=ObjectStatus.ACTIVE).select_related("organization"),
         step=100,
     ):
+        project_settings = get_performance_issues_project_settings(project.id)
```

Comment:
Is this going to result in some kind of n+1 like issue where we check the options db for every project one by one?

Reply:
To handle this in batches, I'd recommend breaking the project iteration into a separate function. Something like

```python
def all_eligible_projects(batch_size=100):
    all_projects = Project.objects.filter(status=ObjectStatus.ACTIVE).select_related("organization")
    for projects in chunked(RangeQuerySetWrapper(all_projects, step=batch_size), batch_size):
        options = ProjectOption.objects.get_value_bulk(projects, "sentry:performance_issue_settings")
        # fill in any default logic here
        for project, option in zip(projects, options):
            yield project, option
```

Reply:
I ended up tweaking this a little bit. We'd need to fetch all projects and iterate through all of them because some projects might have transaction regression issues turned off but function regression issues still enabled